libcron/include/libcron/Task.h

```diff
@@ -1,26 +1,56 @@
 #pragma once
 
 #include <functional>
-#include "CronData.h"
-#include "CronSchedule.h"
 #include <chrono>
 #include <utility>
+#include "CronData.h"
+#include "CronSchedule.h"
 
 namespace libcron
 {
-    class Task
+    class TaskInformation
+    {
+        public:
+            virtual ~TaskInformation() {}
+            virtual std::chrono::system_clock::duration get_delay() const = 0;
+    };
+
+    class Task : public TaskInformation
     {
         public:
+            using TaskFunction = std::function<void(const TaskInformation*)>;
```

Comment:
Make it a reference instead, removes the need to check for nullptr in callback.

Reply:
Is this possible if `TaskInformation` is pure virtual? You mean `const TaskInformation&`? At least I cannot get it compiled as TaskInformation is pure virutal..... Or is there something I am missing?

Reply:
Ah, okay, I found my mistake.... `auto` does not handle references..... `auto&` does work:
`cron.add_schedule("validate schedule", schedule, [](auto&) {});`

Reply:
Done!